src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -136,6 +139,47 @@ public void Dispose()
         public DataFrame Table(string tableName) =>
             new DataFrame((JvmObjectReference)_jvmObject.Invoke("table", tableName));
 
+        /// <summary>
+        /// Returns a dataframe as per the schema and data.
+        /// </summary>
+        /// <param name="data">List of Row objects</param>
+        /// <param name="schema">Schema as StructType</param>
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<GenericRow> data, StructType schema) =>
+            new DataFrame((JvmObjectReference)_jvmObject.Invoke("createDataFrame", data, DataType.FromJson(_jvmObject.Jvm, schema.Json)));
+
+        /// <summary>
+        /// Returns a dataframe when given only the data and no schema
+        /// </summary>
+        /// <param name="data">List of Row objects</param>        
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<GenericRow> data)
+        {
+            var enumerator = data.GetEnumerator();
+            enumerator.MoveNext();
+            var colIndex = 1;
+            var schemaFields = new List<StructField>();
+
+            foreach (object value in enumerator.Current.Values)
+            {
+                if (value.GetType() == typeof(int))
+                {
+                    var field = new StructField("_" + colIndex.ToString(), new IntegerType());
+                    schemaFields.Add(field);
+                }
+                else if (value.GetType() == typeof(string))
+                {
+                    var field = new StructField("_" + colIndex.ToString(), new StringType());
+                    schemaFields.Add(field);
+                }
+                //TODO: Add support for other types                
+                colIndex += 1;
+            }
```

Comment:
Let's rewrite this.  Something like:

```csharp
object[] values = data.First().Values;
for (int i = 0; i <= values.Length; ++i)
{
    switch(values[i])
    {
        ...
    }
}
```

Reply:
data.First() throws an error but incorporated other changes